app/controllers/hq/users_controller.rb

```diff
@@ -6,4 +6,39 @@ def index
     @users = User.all
   end
 
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      flash[:success] = 'User successfully saved'
+      redirect_to hq_user_url(@user)
+    else
+      render 'edit'
+    end
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new
+    @user.attributes = user_params
+    if @user.save
+      flash[:success] = 'User successfully created'
+      redirect_to hq_user_url(@user)
+    else
+      render 'new'
+    end
+  end
+
+private
+
+  def user_params
+    params[:user].permit(:user_name, :email)
```

Comment:
:+1: `params.require(...).permit(...)` is a useful Rails convention. Prefer to use it unless there is good reason against it. Partners controller has been updated to this style as well.